### PR TITLE
region lib updates for CIAO 4.14

### PR DESCRIPTION
**Pull request by wmclaugh**
updates to fold in changes to CIAO region lib for CIAO 4.14. Specifically, fixes for SM-89: bug in pie shape (extent/inside logic), SL-243: region - deprecate obsoleted Warning.

### CODE REVIEW

**Comment by wmclaugh** [comment]
**Summary**

Updates the extern/region library dependency to include the region changes that will be released with CIAO 4.14.

**Details**

This change updates the region code that is included in stand alone sherpa to match the version of the region library which will be released in CIAO 4.14.  The changes include fixes to two issues:

    [SL-243: region - deprecate obsoleted Warning.](https://cxcjira.cfa.harvard.edu/jira/browse/SL-243)

        Removed` warning message "Zero length polygon line segment at ([x], [y])" from regPolygon as this message is no longer meaningful.

    [SM-89: bug in pie shape (extent/inside logic)](https://cxcjira.cfa.harvard.edu/jira/browse/SM-89)

        Corrected Pie Shape area/extent calculations when angles fall outside the bounds range of 0:360.

The extern/region-4.13 directory was renamed to extern/region-4.14 for clarity.

File: extern/region-4.14/Makefile.in [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# Makefile.in generated by automake 1.16.1 from Makefile.am.
```
Comment (DougBurke) [inline extern/region-4.14/Makefile.in:1]: @wmclaugh - do we care that we are now using a significantly older `automake` than was used previously (I think I've asked this before when @olaurino updated some of these dependencies)?

Comment (wmclaugh) [reply extern/region-4.14/Makefile.in]: @DougBurke - thanks for catching this. I've updated the PR to utilize 1.16.1 for consistency.

**Comment by DougBurke** [comment]
@wmclaugh - are you looking for approval for this, or just waiting on other things (as it's not something that I think needs SDS approval)?

**Comment by wmclaugh** [comment]
@DougBurke - It would be good to have it approved- always nice to have a second set of eyes check it out.

File: extern/region-4.14/configure [hunk 2]
```diff
@@ -4304,8 +4284,8 @@ esac
 
 
 
-macro_version='2.4.6'
-macro_revision='2.4.6'
+macro_version='2.4.2'
+macro_revision='1.3337'
```
Comment (DougBurke) [inline extern/region-4.14/configure:4288]: From https://github.com/sherpa/sherpa/pull/1204/files?file-filters%5B%5D=.ac&file-filters%5B%5D=.am&file-filters%5B%5D=.c&file-filters%5B%5D=.h&file-filters%5B%5D=.in&file-filters%5B%5D=.l&file-filters%5B%5D=.m4&file-filters%5B%5D=.org&file-filters%5B%5D=.sh&file-filters%5B%5D=.y&file-filters%5B%5D=No+extension#diff-f22710e4eb0fbe749656c275a5f14cc05805cf6c90afa013f0216809f80f7586 it looks like we've regressed the "ltmain" version from 2,4,6 to 2.4.2

From what I can tell it's not a problem as long as we include the libtool.m4 files (and probably others)  of the correct version (which we do) but looking at the details of some of these files they deal with OS system versions which makes me wonder if we would want an update of these for the Apple M1 platform (or is that not needed)?

**Comment by DougBurke** [comment]
For (my) reference, https://github.com/sherpa/sherpa/pull/598 is the last time I remember we had this issue

File: extern/region-4.14/configure (hunk 2 above)
Comment (DougBurke) [reply extern/region-4.14/configure:4288]: I also wonder if it's an issue for aarch64 support #1186 - which is not a high priority for us but the community may appreciate it?

File: extern/region-4.14/ltmain.sh [hunk 3]
```diff
@@ -64,7 +64,7 @@ package_revision=2.4.6
 # libraries, which are installed to $pkgauxdir.
 
 # Set a version string for this script.
-scriptversion=2015-10-12.13; # UTC
+scriptversion=2015-01-20.17; # UTC
```
Comment (DougBurke) [inline extern/region-4.14/ltmain.sh:67]: I'm going to note that this appears to be a reversion but it looks like this is due to `2.4.6 Debian 2.4.6-4` has changed to `2.4.6` so I'm not going to ask for this to be updated.

**Review by DougBurke — APPROVED** [review]
I noted a small difference in the libtool version but I don't think it's significant.

I do note that the latest autoconf version is 2.71 but I assume that we don't need to update to that yet.

**Pull request merged**